### DISCUSSION TITLE
Android support

### DISCUSSION THREAD

**Discussion by najmsheikh** [itsToggle/plex_debrid#43] [Q&A] [answered]
Has anyone been successful in running a Plex _server_ on their Android device? I was able to succesfully mount the debrid drive and can access the files via terminal and other media players on the phone but Plex cannot seem to detect the mounted drive. For clarity's sake, I mounted directly to `sdcard/Download` so it should have showed up in the downloads folder. Any thoughts?

**Comment by itsToggle** [comment, marked as answer]
Have you tried adding the mounting tag '--allow-other'? I sadly cant confirm this on android, but on my Linux VPS this was necessary

**Reply by najmsheikh** [reply]
> Have you tried adding the mounting tag '--allow-other'? I sadly cant confirm this on android, but on my Linux VPS this was necessary

Yep, that was at least necessary to allow other apps to see the files at first. Plex still cannot index the files however.

Here's the full command:
```
rclone --config sdcard/.rclone.conf mount debrid-remote:links /sdcard/Download --dir-cache-time=10s --vfs-cache-mode=full --allow-non-empty --allow-other --cache-dir=/sdcard/debrid-cache
```

**Reply by itsToggle** [reply]
I dont think it'll make a differencd, but you could try to remove the 'config' command, since 'config' and 'mount' are different commands.

--alllow-non-empty should be removed aswell as this command allows for overwriting existing mounts - which you shouldnt need.

What you could try to add is the 'unmask' tag, for which I would refer you to the official rclone documentation. I think the tag would look something like this: '--unmask 000' but im not sure.

Another option would be to make sure the linux user that Plex is using can access the mounted directory. I cant find a resource for this right now, but you should be able to find some more information by google searching this issue.

Edit: Here is something a reddit user did to make the files visible on Linux: https://www.reddit.com/r/Piracy/comments/v5zpj7/comment/igl36eo/?context=3

**Reply by itsToggle** [reply]
Here is a tutorial to mount gdrive on an android shield - they are using a second 'bind' command - perhaps thats whats missing: https://forums.plex.tv/t/howto-configuring-rclone-mount-on-nvidia-shield-to-replace-plex-cloud/371318

**Reply by itsToggle** [reply]
Okay after a bit more searching, I would recommend you use the rclone magisk module: https://github.com/Magisk-Modules-Repo/com.piyushgarg.rclone and mount realdebrid through their webdav implementation.

**Reply by najmsheikh** [reply]
Thanks again so much for looking into this! I was ultimately able to pinpoint the issue being with Plex Media Server itself instead of with the rclone setup. I'm running the server on a Pixel 3a which isn't officially supported by Plex so I sideloaded the apk from [here](https://www.apkmirror.com/apk/plex-inc/plex-media-server/). Ultimately what resolved it was starting, resetting, restarting the server a few times. Eventually I was able to get the Media Server to show up under my account. It continued to have some connectivity issues during library setup but eventually got it working.

I also played around with Magisk's mounting settings. I tried both global namespacing and inherited. It didn't _seem_ to make any difference in my case but just FYI I currently have it set to inheritance mode.

For the rclone Magisk module, I think the one you linked might be slightly outdated so I used [this one](https://github.com/AvinashReddy3108/rclone-mount-magisk) instead. 

> I dont think it'll make a differencd, but you could try to remove the 'config' command, since 'config' and 'mount' are different commands.

The snippet I linked above is still mounting. The reason I added the `--config` and `--cache-dir` arguments is because by default rclone tries to write to root `/` directory which it does not have permission to. Changing where the config and cache are simply allows rclone to work as expected.

I used `--allow-non-empty` because I was under the impression that the Android version of Plex Media Server simply scans the Download / DCIM / Movies folder(s) for media. Plex, _the client app_, does but Plex _Media Server_ actually allows you to define directories just as you would on other platforms. So yes, on future runs the `--allow-non-empty` argument won't be necessary.

I didn't need to bind or use the `--umask` argument. The key was largely `--allow-other` and tinkering with the Media Server to kickstart the server onboarding process.

The script is working fine for now. I did notice a bug with the AD integration where cached links are added to `magnets` instead of `links` directory; but i suspect that could just be due to the Android script being a couple commits behind master. I'll see what's different in the android branch vs master.
Sometime soon I might submit a pull request to make this entirely into a Magisk module :)

**Reply by itsToggle** [reply]
Awesome! Great that its working now. Are you experiencing any compatability issues with the "ported" plex media server apk? I don't own a shield and was thinking of rooting one of my old phones for a setup like yours :)

The android branch is not actually needed anymore, you can simply clone the master branch. I used to have them in different branches because of some module incompatibility, but that has been resolved. I will update the android branch instructions, totally forgot about it :)

Turning this into a magisk module sounds awesome! :)

**Reply by najmsheikh** [reply]
I wanted to provide an update since this thread seems to be the primary one for Android discussion. While this Plex integration works, I personally wouldn't recommend this setup. This is no fault of this project but rather Plex Media Server itself. I constantly find myself having to restart the phone for PMS to find the new content which is quite a hassle. I imagine these issues likely don't exist on an Nvidia SHIELD (which PMS Android was designed for) but I don't have a Shield to verify with.

**Reply by itsToggle** [reply]
Oh what a bummer.. :/ 
[Here](https://www.reddit.com/r/PleX/comments/jr3z2x/new_files_are_not_transferring_to_plex_media/) are some people who expirienced the same type of issue on their shield with an HDD, they seem to have solved it by cleanly re-installing the Plex Server and Client - have you tried this? 
Could also be an rclone for android specific issue, I'll check if I can find anything.
